docs/docs/concepts/overview.md

```diff
@@ -0,0 +1,194 @@
+---
+sidebar_position: 1
+title: Architecture Overview
+---
+
+# Architecture Overview
+
+Reflector is built as a modern, scalable, microservices-based application designed to handle audio processing workloads efficiently while maintaining data privacy and control.
+
+## System Components
+
+### Frontend Application
+
+The user interface is built with **Next.js 14** using the App Router pattern, providing:
+
+- Server-side rendering for optimal performance
+- Real-time WebSocket connections for live transcription
+- WebRTC support for audio streaming
+- Responsive design with Chakra UI components
+
+### Backend API Server
+
+The core API is powered by **FastAPI**, a modern Python framework that provides:
+
+- High-performance async request handling
+- Automatic OpenAPI documentation generation
+- Type safety with Pydantic models
+- WebSocket support for real-time updates
+
+### Processing Pipeline
+
+Audio processing is handled through a modular pipeline architecture:
+
+```
+Audio Input → Chunking → Transcription → Diarization → Post-Processing → Storage
+```
+
+Each step can run independently and in parallel, allowing for:
+- Scalable processing of large files
+- Real-time streaming capabilities
+- Fault tolerance and retry mechanisms
+
+### Worker Architecture
+
+Background tasks are managed by **Celery** workers with **Redis** as the message broker:
+
+- Distributed task processing
+- Priority queues for time-sensitive operations
+- Automatic retry on failure
+- Progress tracking and notifications
+
+### GPU Acceleration
+
+ML models run on GPU-accelerated infrastructure:
+
+- **Modal.com** for serverless GPU processing
+- Support for local GPU deployment (coming soon)
+- Automatic scaling based on demand
+- Cost-effective pay-per-use model
+
+## Data Flow
+
+### File Processing Flow
+
+1. **Upload**: User uploads audio file through web interface
+2. **Storage**: File stored temporarily or in S3
+3. **Queue**: Processing job added to Celery queue
+4. **Chunking**: Audio split into 30-second segments
+5. **Parallel Processing**: Chunks processed simultaneously
+6. **Assembly**: Results merged and aligned
+7. **Post-Processing**: Summary, topics, translation
+8. **Delivery**: Results stored and user notified
```

Comment:
We do not split into 30 second segments anymore, this whole section should be rewritten to match the current daily behavior